### DISCUSSION TITLE
Start Hyprland locally, but from ssh

### DISCUSSION THREAD

**Discussion by cjuniorfox** [hyprwm/Hyprland#3201] [Q&A]
I'm willing to log in through SSH at the computer and start Hyprland locally.
I do not want to share the screen remotely but trigger the session at the physical display adapter from the SSH.
From the old times of X, achieved that by assigning the DISPLAY=:0 variable.
```
DISPLAY=:0 exec i3
```
What I'm missing here?

Trying to do the same but with Hyprland, I got the following log output:
```
$ DISPLAY=:0 exec Hyprland
Welcome to Hyprland!
Instance Signature: 96d555e8e794627bfc561e294e148ab8a9961fcc_1694035513
Hyprland PID: 3444
===== SYSTEM INFO: =====
System name: Linux
Node name: zentac
Release: 6.4.13-200.fc38.x86_64
Version: #1 SMP PREEMPT_DYNAMIC Wed Aug 30 17:07:31 UTC 2023


GPU information:
07:00.0 VGA compatible controller [0300]: NVIDIA Corporation GA104 [GeForce RTX 3070] [10de:2484] (rev a1) (prog-if 00 [VGA controller])


Warning: you're using an NVIDIA GPU. Make sure you follow the instructions on the wiki if anything is amiss.

os-release:
NAME="Fedora Linux"
VERSION="38 (Thirty Eight)"
ID=fedora
VERSION_ID=38
VERSION_CODENAME=""
PLATFORM_ID="platform:f38"
PRETTY_NAME="Fedora Linux 38 (Thirty Eight)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:38"
DEFAULT_HOSTNAME="fedora"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f38/system-administrators-guide/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=38
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=38
SUPPORT_END=2024-05-14

========================



If you are crashing, or encounter any bugs, please consult https://wiki.hyprland.org/Crashes-and-Bugs/



Current splash: Never gonna let you down (I am trying!)


Creating the HookSystem!
Creating the KeybindManager!
Creating the AnimationManager!
Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 22.29µs. Estimated average calc time: 0.05µs.
Creating the ConfigManager!
Creating the CHyprError!
 [hookSystem] New hook event registered: focusedMon
 [hookSystem] New hook event registered: preRender
Creating the LayoutManager!
Reloading the config!
Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 9.71µs. Estimated average calc time: 0.04µs.
Using config: /home/junior/.config/hypr/hyprland.conf
Created a bezier curve, baked 255 points, mem usage: 4.08kB, time to bake: 10.59µs. Estimated average calc time: 0.03µs.
Registered dynamic var "$mainMod" -> SUPER
00:00:00.000 [backend/x11/backend.c:396] Creating X11 backend
00:00:00.000 [backend/x11/backend.c:409] Failed to open xcb connection
terminate called after throwing an instance of 'std::runtime_error'
  what():  wlr_backend_autocreate() failed!
Hyprland has crashed :( Consult the crash report at /home/junior/.hyprland/hyprlandCrashReport3444.txt for more information.
```
I appreciate any help by advance.

**Comment by memchr** [comment]
I don't think it's trivial, I used to remotely debug Hyprland on a VM by getting getty to automatically start a lldb server.

I think you can achieve a similar effect with ssh

- use `chvt <vt number>` to switch to a tty. 
- Set getty to auto login on that tty.

**Reply by cjuniorfox** [reply]
Already tried. Didn´t work :(

**Reply by memchr** [reply]
```
systemctl cat getty@tty1.service
```

**Reply by memchr** [reply]
Sorry, Does the above seem unclear?

Execute
```bash
sudo systemctl edit getty@tty1.service
# you can change getty@tty1 to whatever tty you like
```
Add the following lines
```ini
[Service]
ExecStart=
ExecStart=-/sbin/agetty -o '-p -f -- \\u' --noclear --autologin arch %I $TERM
```
Save and quit your editor.

Then open your `~/.bashrc`, append the following lines
```bash
if fgconsole &>/dev/null; then
	Hyprland
fi
```
Save and quit your editor, then run `systemctl soft-reboot` to check the effect.
In the ssh session, execute
```bash
chvt 1
# or whatever tty you have configured
```
To launch the compositor.

**Comment by vaxerski** [comment]
```
DISPLAY=:0 exec i3
```
Maybe because i3 just connects to a running Xorg instance that was started on the actual computer, while Hyprland is like Xorg itself?

what are you trying to do???

There are many things that will go wrong here. Seatd, drm, kms... You are better off starting Hyprland on your machine (either through getty or something) and then ssh-ing to that.

**Reply by cjuniorfox** [reply]
I'm trying to remotely start a local Hyprland session.
As many as those wrong things are happening, the Hyprland is blabbing about the seats, but I really don't know.

**Comment by theKlanc** [comment]
Just `sudo seatd -u $USER -g $USER` and then run `hyprland` 
If you need to start a graphical program from ssh set `export WAYLAND_DISPLAY=wayland-1`, and get the instance signature like this `export HYPRLAND_INSTANCE_SIGNATURE=$(cat $XDG_RUNTIME_DIR/hypr/$(ls -t $XDG_RUNTIME_DIR/hypr/ | head -n 1)/hyprland.log | grep "Instance Signature" | cut -d' ' -f 4 | head -n1)`
And you'll be able to start any app as if you were inside hyprland itself.
